engine/src/main/java/org/terasology/engine/ComponentSystemManager.java

```diff
@@ -128,6 +128,7 @@ public void register(ComponentSystem object) {
         context.get(EntityManager.class).getEventSystem().registerEventHandler(object);
 
         if (initialised) {
+            logger.warn("System " + object.getClass().getName() + " registered post-init.");
```

Comment:
Why is this a warning? Should this be avoided? Why?

Reply:
System has two methods - `preInit` and `postInit`. The javadoc for the later actually even guarantees that all systems have already been registered by the time it has been called. So, when something is registered later, that should definitely not happen